ArduCopter/AP_Arming.cpp

```diff
@@ -633,7 +633,14 @@ bool AP_Arming_Copter::arm_checks(AP_Arming::Method method)
         SRV_Channels::set_emergency_stop(false);
         // if we are using motor Estop switch, it must not be in Estop position
     } else if (rc().find_channel_for_option(RC_Channel::AUX_FUNC::MOTOR_ESTOP) && SRV_Channels::get_emergency_stop()){
-        gcs().send_text(MAV_SEVERITY_CRITICAL,"Arm: Motor Emergency Stopped");
+        check_failed(true, "Motor Emergency Stopped");
+        return false;
+    }
+
+    // if we are using Disarm switch and it's asserted, fail to arm
+    RC_Channel *chan = rc().find_channel_for_option(RC_Channel::AUX_FUNC::DISARM);
+    if (chan != nullptr && chan->get_aux_switch_pos() == RC_Channel::aux_switch_pos_t::HIGH){
```

Comment:
This code now appears to be duplicated in AP_Arming... should be removed here?

Reply:
Thank you!
I learned a lot.